File: src/main/scala/com/vc/easy/L111.scala
package com.vc.easy


class TreeNode111(var _value: Int) {
  var value: Int = _value
  var left: TreeNode111 = null
  var right: TreeNode111 = null
}

object L111 {
  def minDepth(root: TreeNode111): Int = {
    if(root == null) 0
    else if(root.left == null) 1 + minDepth(root.right)
    else if(root.right == null) 1 + minDepth(root.left)
    else 1 + Math.min(minDepth(root.left), minDepth(root.right))
  }
}

File: src/main/java/com/vc/easy/L111.java
package com.vc.easy;

import com.vc.hard.TreeNode;

class L111 {
    public int minDepth(TreeNode root) {
        if(root == null) return 0;

        int minDepth = Integer.MAX_VALUE;
        if(root.left != null) {
            minDepth = Math.min(minDepth, minDepth(root.left));
        }

        if(root.right != null) {
            minDepth = Math.min(minDepth, minDepth(root.right));
        }

        return 1 + (minDepth == Integer.MAX_VALUE ? 0 : minDepth);
    }
}
